### PR TITLE
Pyaction use actionx

### PR DESCRIPTION
**Pull request by joakim-hove**
Continuation of: https://github.com/OPM/opm-common/pull/2932

Downstream: 
https://github.com/OPM/opm-simulators/pull/3787
https://github.com/OPM/opm-tests/pull/714

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3787 opm-tests=714 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3787 opm-tests=714 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3787 opm-tests=714 please

File: tests/msim/MSIM_PYACTION_ACTIONX.DATA [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+
+-- Copyright (C) 2015 Statoil
+
+-- This simulation is based on the data given in 
+-- 'Comparison of Solutions to a Three-Dimensional
+-- Black-Oil Reservoir Simulation Problem' by Aziz S. Odeh,
+-- Journal of Petroleum Technology, January 1981
```
Comment (blattms) [inline tests/msim/MSIM_PYACTION_ACTIONX.DATA:11]: I think you should briefly mention what was changed.

**Review by blattms — COMMENTED** [review]
Of course I am not very familiar with this code, but it looks good to apart from the following comments:

I think the documentation should move from the *.cpp (e.g. Schedule.cpp) to the headers, as that is developers would search and doxygen can be made aware of.

**Comment by joakim-hove** [comment]
Regarding documenration/comments: I feel the level of detail here means that it makes most sense to have it in the cpp file - this is not for *users* of the api, but rather developers working with the actual implementation?

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3787 opm-tests=714 please

**Pull request merged**